scripts/setup-ubuntu.sh

```diff
@@ -150,6 +150,9 @@ function install_duckdb {
     (
       cd duckdb
       cmake_install -DBUILD_UNITTESTS=OFF -DENABLE_SANITIZER=OFF -DENABLE_UBSAN=OFF -DBUILD_SHELL=OFF -DEXPORT_DLL_SYMBOLS=OFF -DCMAKE_BUILD_TYPE=Release
+      # duckdb build process creates ccache dir with root ownership if dir does not exist
+      # Change ccache dir ownership to current user to avoid permission issues
+      sudo chown -R $(id -u -n):$(id -g -n) $(ccache -k cache_dir)
```

Comment:
Same here, I think the user should set either `CCACHE_DIR=...` envvar or have a ccache config. The script modifying permissions feels intrusive.

Reply:
Do you mean set a different `CCACHE_DIR` when install duckdb ?

Reply:
I think we should get the ccache dir with `ccache -k cache_dir` check if it exists and if it doesn't exists create it without sudo to avoid the issue with duckdb. That way if the user is already a ccache user potentially with a working config etc. we don't have to touch anything and we also don't have to use chown

Reply:
`ccache` creates subdirectories under cache dir
<img width="757" alt="image" src="https://github.com/facebookincubator/velox/assets/88070094/64f923e0-556e-442b-9154-6e45f9467a13">
Subsequence make processes will encounter `Permission denied error` when visiting these subdirectories.

Reply:
@assignUser How about set `CCACHE_DIR ` to `$(ccache -k cache_dir)/duckdb` when installing duckdb?

Reply:
Hm, I think the root issue is `cmake_install` with `  ${SUDO} ninja -C "${BINARY_DIR}" install`.
This causes ccache to be launched as root. If we split build and install it should resolve this entire issue, could you test replacing the ninja line with: `cmake --build "${BINARY_DIR}"` and `${SUDO} cmake --install "${BINARY_DIR}"`

Reply:
(sorry I could have come to this conclusion earlier)

Reply:
@assignUser Never mind :). Let me try.

Reply:
@assignUser 
> could you test replacing the ninja line with: cmake --build "${BINARY_DIR}" and ${SUDO} cmake --install "${BINARY_DIR}"

This works and I have updated the code. Thanks for your guidance.

Reply:
Ah great, I will start the CI!